### PR TITLE
[vcpkg] Add option cmake-args 

### PR DESCRIPTION
**Pull request by Neumann-A**
to pass additional args to the cmake script execution within vcpkg

things users/maintainers might want to pass:
--trace/--trace-expand
CMAKE_MESSAGE_LOG_LEVEL
CMAKE_MESSAGE_INDENT
CMAKE_MESSAGE_CONTEXT_SHOW/ --log-context (If vcpkg would use CMAKE_MESSAGE_CONTEXT)
PORT_DEBUG=ON

### CODE REVIEW

**Comment by ras0219** [comment]
This should probably imply/require --editable.

**Comment by Neumann-A** [comment]
@ras0219 This is not passed to the `vcpkg_configure_cmake`. 
This is added to `[DEBUG] CreateProcessW("E:\vcpkg_cache\downloads\tools\cmake-3.17.2-windows\cmake-3.17.2-win32-x86\bin\cmake.exe" <alot of cmake_args> -P "E:\qt_zstd\scripts\ports.cmake")`

**Comment by ras0219** [comment]
Sure, but the point is that these options can change the behavior of the package build -- that means they shouldn't be used as a basis for a cached binary tree.

**Comment by Neumann-A** [comment]
Then there needs to be a whitelist for cmake-args which don't imply --editable. (Mainly those vars mentioned in the very first comment.)

**Comment by strega-nil** [comment]
We had a discussion about this at today's meeting, and our opinion is that:

1. The option needs to be renamed `--x-cmake-args`
2. This needs to imply `--editable` for now, and we can special case stuff afterwards.

Once those are in, we're up to merge.

**Comment by Neumann-A** [comment]
@strega-nil Done

File: toolsrc/src/vcpkg/install.cpp [hunk 1]
```diff
@@ -759,7 +762,7 @@ namespace vcpkg::Install
         const bool no_downloads = Util::Sets::contains(options.switches, (OPTION_NO_DOWNLOADS));
         const bool only_downloads = Util::Sets::contains(options.switches, (OPTION_ONLY_DOWNLOADS));
         const bool is_recursive = Util::Sets::contains(options.switches, (OPTION_RECURSE));
-        const bool is_editable = Util::Sets::contains(options.switches, (OPTION_EDITABLE));
+        const bool is_editable = Util::Sets::contains(options.switches, (OPTION_EDITABLE)) | !args.cmake_args.empty();
```
Comment (Neumann-A) [inline toolsrc/src/vcpkg/install.cpp:765]:
```suggestion
        const bool is_editable = Util::Sets::contains(options.switches, (OPTION_EDITABLE)) || !args.cmake_args.empty();
```

File: toolsrc/include/vcpkg/vcpkgcmdarguments.h [hunk 2]
```diff
@@ -149,6 +149,9 @@ namespace vcpkg
         constexpr static StringLiteral BINARY_SOURCES_ARG = "binarysource";
         std::vector<std::string> binary_sources;
 
+        constexpr static StringLiteral CMAKE_SCRIPT_ARG = "x-cmake-args";
```
Comment (strega-nil) [inline toolsrc/include/vcpkg/vcpkgcmdarguments.h:152]: You'll also need to add a definition for this in `vcpkgcmdarguments.cpp` for VS2015 and GCC 6.

**Comment by strega-nil** [comment]
Thanks @Neumann-A

**Pull request merged**